lib/output.js

```diff
@@ -1964,19 +1964,14 @@ function OutputStream(options) {
     function make_num(num) {
         var str = num.toString(10), a = [ str.replace(/^0\./, ".").replace("e+", "e") ], m;
         if (Math.floor(num) === num) {
-            if (num >= 0) {
-                a.push("0x" + num.toString(16).toLowerCase(), // probably pointless
-                       "0" + num.toString(8)); // same.
-            } else {
-                a.push("-0x" + (-num).toString(16).toLowerCase(), // probably pointless
-                       "-0" + (-num).toString(8)); // same.
-            }
-            if ((m = /^(.*?)(0+)$/.exec(num))) {
+            a.push(
+                (num >= 0 ? "0x" : "-0x") + num.toString(16).toLowerCase() // probably pointless
+            );
+            if ((m = /^(.*?)(0{3,})$/.exec(num))) {
                 a.push(m[1] + "e" + m[2].length);
             }
         } else if ((m = /^0?\.(0+)(.*)$/.exec(num))) {
-            a.push(m[2] + "e-" + (m[1].length + m[2].length),
-                   str.substr(str.indexOf(".")));
```

Comment:
Just to understand correctly: we don't need to consider `str.substr(str.indexOf("."))`, since that is already handled above by `str.replace(/^0\./, ".")`. Correct?

Reply:
Correct